### PR TITLE
Support @ValidatesRunner(RunnableOnService) in Python [2/2]

### PR DESCRIPTION
**Pull request by markflyhigh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

 - Create TestPipeline class which is a wrapper of Pipeline to construct beam pipeline and will only used in test (ValidatesRunner test and E2E integration test).
 - Provide a small set of examples that how to build ValidatesRunner tests.
 - Currently, the default runner is DirectRunner if it's not specified in command line.

Test done by running all unit test + ValidatesRunner test locally. 
A subset of ValidatesRunner tests run successfully on Dataflow service.

### CODE REVIEW

**Comment by markflyhigh** [comment]
+R: @aaltay @robertwb

File: sdks/python/apache_beam/test_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  def create_pipeline_opt_from_args(self):
+    """Create a pipeline options from command line argument: --test-options"""
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--test-options',
+                        type=str,
+                        action='store',
+                        help='only run tests providing service options')
+    known, unused_argv = parser.parse_known_args()
+
+    options = []
```
Comment (robertwb) [inline sdks/python/apache_beam/test_pipeline.py]: Nit: prefer single assignment (e.g.
```
if known.test_options:
  ...
else:
  options = []
```
) or better let the default value do the right thing without the need for special casing it.

File: sdks/python/apache_beam/test_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    if known.test_options:
+      options = known.test_options.split()
```
Comment (robertwb) [inline sdks/python/apache_beam/test_pipeline.py]: What about arguments that contain whitespace? Use `shlex.split` instead.

Comment (markflyhigh) [inline sdks/python/apache_beam/test_pipeline.py]: done

**Comment by markflyhigh** [comment]
+R: @tgroh

**Comment by aaltay** [comment]
@markflyhigh is this ready to be merged, are test errors unrelated?

**Comment by markflyhigh** [comment]
Jenkins error is related to java integration test failure which is unrelated to Python. Will rebase and rerun Jenkins build.

**Comment by aaltay** [comment]
Thank you. Please ping @tgroh  when this is ready to merge.

**Pull request closed without merging**